base/deprecated.jl

```diff
@@ -3806,6 +3806,32 @@ end
     @deprecate getq(F::Factorization) F.Q
 end
 
+@deprecate method_exists(f, t, world) method_exists(f, t, world = world)
+@deprecate Timer(timeout, repeat) Timer(timeout, repeat = repeat)
+@deprecate Timer(callback, delay, repeat) Time(callback, delay, repeat = repeat)
+@deprecate names(m, all) names(m, all = all)
+@deprecate names(m, all, imported) names(m, all = all, imported = imported)
+@deprecate code_native(io, f, types, syntax) code_native(io, f, types, syntax = syntax)
+@deprecate code_native(f, types, syntax) code_native(f, types, syntax = syntax)
+@deprecate eachmatch(re, str, overlap) eachmatch(re, str, overlap = overlap)
+@deprecate matchall(re, str, overlap) matchall(re, str, overlap = overlap)
+@deprecate chop(s, head) chop(s, head = head)
+@deprecate chop(s, head, tail) chop(s, head = head, tail = tail)
+@deprecate tryparse(T, s, base) tryparse(T, s, base = base)
+@deprecate parse(T, s, base) parse(T, s, base = base)
+@deprecate reverse!(v, start) reverse!(v, start = start)
+@deprecate reverse!(v, start, stop) reverse!(v, start = start, stop = stop)
+@deprecate reverse(v, start) reverse(v, start = start)
+@deprecate reverse(v, start, stop) reverse(v, start = start, stop = stop)
```

Comment:
I wonder whether `reverse` should be kept as is. There's a pattern of passing start and stop indices as keyword arguments, e.g. with `SubString` and `copy!`. I prefer using keyword arguments for arguments which are of a "different nature" than other arguments (typically, options vs. data).

Reply:
I don't understand...

Reply:
What don't you understand exactly? :-)

Reply:
@nalimilan did you typo "There's a pattern of passing start and stop indices as keyword argument"?  I think you meant ~keyword~ positional argument.

Reply:
Ah, thanks, that explains it!

Reply:
I would be fine with leaving `reverse` and `reverse!` as they are but I don't hate this either.

Reply:
I'm not super passionate either but its worth noting that if they are both optional arguments than you wouldn't be able to do something like `reverse(v, stop = stop)`

Reply:
Yeah, you would have to pass `start(v)` before the stop index. But this remark also applies to functions with similar arguments, so I think we should keep them consistent (either way).

Reply:
Do you have a list of such functions? It might be just worth changing all of them (if they are exported).

Reply:
How about dropping that part for now so that we can get this merged now?